### PR TITLE
Fixes and minor cleanup

### PR DESCRIPTION
**Pull request by Fraktality**
Conglomeration of small tweaks coupled with a few fixes.

Here are the most notable changes:
- Fixed character mass calculation
   - You can safely lose limbs now
- Improved castCylinder's point sample distribution
   - You won't see the difference
- Improved numerical stability of the move direction calculation when the camera looks straight up or down

This PR is all over the place; let me know if you need it split up.

### CODE REVIEW

**Comment by Validark** [comment]
Nice work!

File: src/core/getModelMass.lua [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
-			mass = mass + child:GetMass()
-		else
-			mass = mass + getModelMass(child)
+	local visited = {}
+	
+	local function sumMass(part)
+		mass = mass + part:GetMass()
+		visited[part] = true
+		
+		for _, p in ipairs(part:GetConnectedParts()) do
```
Comment (jovannic) [inline src/core/getModelMass.lua]: This should probably be (recursive = true)

File: src/core/castCylinder.lua [hunk 2]
```diff
@@ -8,22 +8,21 @@ local CAST_COUNT = 32
...
-	for i, p in pairs(points) do
-		local w = weights[i]
-		n = n + w
-		sumX, sumY, sumZ = sumX + p.X*w, sumY + p.Y*w, sumZ + p.Z*w
+	local pointSum = Vector3.new()
+	local weightSum = 0
+	for i, point in pairs(points) do
+		local weight = weights[i]
+		weightSum = weightSum + weight
+		pointSum = pointSum + point*weight
```
Comment (jovannic) [inline src/core/castCylinder.lua]: This allocates 2 new vectors for each loop. That's an additional 64 trash vector allocs per luanoid per heatbeat.

Might be premature to worry about garbage allocs, but... thoughts?

File: src/core/CharacterState/Walking.lua [hunk 3]
```diff
@@ -67,43 +65,35 @@ local function createForces(character)
...
 	-- orthonormalize, keeping up vector
-	looku = -upu:Cross(rightu).Unit
-	return CFrame.new(0, 0, 0, rightu.x, upu.x, looku.x, rightu.y, upu.y, looku.y, rightu.z, upu.z, looku.z)
+	rZ = rX:Cross(rY).Unit
+	return CFrame.new(
+		0, 0, 0,
+		rX.x, rY.x, rZ.x,
+		rX.y, rY.y, rZ.y,
+		rX.z, rY.z, rZ.z
+	)
```
Comment (jovannic) [inline src/core/CharacterState/Walking.lua]: 👍

File: src/core/CharacterState/Walking.lua [hunk 4]
```diff
@@ -67,43 +65,35 @@ local function createForces(character)
...
-		angle = angle - THETA
-	end
-	return angle
-end
-
 local function angleShortest(a0, a1)
-	local d1 = angleAbs(a1 - a0)
-	local d2 = -angleAbs(a0 - a1)
-	return math.abs(d1) > math.abs(d2) and d2 or d1
+	return (a1 - a0 + math.pi)%(2*math.pi) - math.pi
```
Comment (jovannic) [inline src/core/CharacterState/Walking.lua]: Ugh. Figured this out before but it was late and I couldn't find it. I am ashamed.

**Pull request merged**